aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/AbortableCountDownLatch.java

```diff
@@ -41,6 +41,13 @@ public void abort(E error) {
         }
     }
 
+    public void abortableAwait() throws InterruptedException, E {
+        super.await();
```

Comment:
⚠️

I think we should use a default timeout. If we wait indefinitely, we may hang the test harness with no clear rationale provided.

Reply:
That's a fair point.  I'm good with leaving the timeout here, for now, but would propose we raise it from 10 seconds to 60 seconds or so.  

This timeout is what limits the number of items that can be synced with `DataStore`, which is bad, so I do think we need a better solution.  Another approach would be to split the timeout into each of the Processors, rather than the `Orchestrator`.  Or perhaps, we could use a watchdog approach, where the timeout is reset whenever a new page of models has been returned from the server.

Reply:
I think the timeouts for the subscription and sync processors might warrant different values. The sync can be very long, but setting up the subscriptions ought to be shorter. I'd always envisioned that we'd make the sync timeout user-configurable. I agree that 10 seconds is too short of a default. 60 does sound like a better starting place. I think our hearts were in the right place with the `(number of model types) * value` approach, but in practice I'm not sure it has helped much.